src/control/server/mgmt_system.go

```diff
@@ -1138,7 +1145,59 @@ func (svc *mgmtSvc) SystemSetProp(ctx context.Context, req *mgmtpb.SystemSetProp
 		return nil, err
 	}
 
-	return &mgmtpb.DaosResp{}, nil
+	// for any system property that has a pool property, update all pools
+	var poolSysProps []*daos.PoolProperty
+	for k := range req.GetProperties() {
+		p, ok := svc.systemProps.Get(k)
```

Comment:
I don't know if this came in from my example (which may have been incomplete), or there was some other misunderstanding, but it doesn't appear that the system-level values are being used when setting the pool-level values.

The `svc.systemProps` field is initialized as a map of prop key strings to the corresponding property objects, but it is not the actual store for the property values. I understand why this may be confusing. We do it this way to decouple the definition of the system property map from the storage of the map in the system db, in order to keep the storage layer more generic and more easily testable.

In any case, unless I'm missing something, you're going to want to get the value here in addition to the key and use it to set the value of the `PoolProperty` here so that the new value is persisted on the pool(s).

Reply:
It does seem to be working the way *I* expected it to. :-D I've manually tested by changing the pool_scrub_mode system property between off and lazy and each time the pool property is updated appropriately. Would system.SetUserProperties update the svc.systemProps map?

Reply:
Spent some time going through the code to understand this. It doesn't seem like it should be updated, since the prop it's manipulating should be a copy. Would like to know at least why it seems to be working, if it's an unexpected side-effect from somewhere else in the code.